JASP-Desktop/mainwindow.cpp

```diff
@@ -524,11 +525,47 @@ void MainWindow::dataSetChanged(DataSet * dataSet)
 }
 
 
+bool MainWindow::hasOldAnalyses()
+{
+	for (int i = 0; i < _analyses->count(); i++)
+	{
+		Analysis *analysis = _analyses->get(i);
+		if (analysis != nullptr && analysis->version() != AppInfo::version)
+			return true;
+	}
+	
+	return false;
+}
+
 
 void MainWindow::setImageBackgroundHandler(QString value)
 {
 	emit imageBackgroundChanged(value);
-	_engineSync->refreshAllPlots();
+
+	if (!hasOldAnalyses())
+	{
+		_engineSync->refreshAllPlots();
+		return;
+	}
+	
+	if (MessageForwarder::showYesNo("Version incompatibility", "Your analyses were created in an older version of JASP, to change the background of the images they must be refreshed first.\n\nRefresh all analyses?"))
+		_analyses->refreshAllAnalyses();
+}
+
+
+void MainWindow::plotPPIChangedHandler(int ppi, bool wasUserAction)
+{
+    emit ppiChanged(ppi);
+
+	if (!hasOldAnalyses())
+	{
+		_engineSync->refreshAllPlots();
+		return;
```

Comment:
It would be nicer to remove the `return` here and change `} if (wasUserAction)` to `else if(wasUserAction && MessageForwarder::showYesNo("Version incompatibility", "Your analyses were created in an older version of JASP, to change the PPI of the images they must be refreshed first.\n\nRefresh all analyses?"))`